File: cpp-tiny/lib/Models/EmojiReactionAllOf.cpp


#include "EmojiReaction_allOf.h"

using namespace Tiny;

EmojiReaction_allOf::EmojiReaction_allOf()
{
	emoji_code = null;
	emoji_name = null;
	reaction_type = null;
	user_id = null;
	user = null;
}

EmojiReaction_allOf::EmojiReaction_allOf(std::string jsonString)
{
	this->fromJson(jsonString);
}

EmojiReaction_allOf::~EmojiReaction_allOf()
{

}

void
EmojiReaction_allOf::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *emoji_codeKey = "emoji_code";
    
    if(object.has_key(emoji_codeKey)) 
    {
        bourne::json value = object[emoji_codeKey];


        

        AnyType* obj = &emoji_code;
		obj->fromJson(value.dump());

    }

    const char *emoji_nameKey = "emoji_name";
    
    if(object.has_key(emoji_nameKey)) 
    {
        bourne::json value = object[emoji_nameKey];


        

        AnyType* obj = &emoji_name;
		obj->fromJson(value.dump());

    }

    const char *reaction_typeKey = "reaction_type";
    
    if(object.has_key(reaction_typeKey)) 
    {
        bourne::json value = object[reaction_typeKey];


        

        AnyType* obj = &reaction_type;
		obj->fromJson(value.dump());

    }

    const char *user_idKey = "user_id";
    
    if(object.has_key(user_idKey)) 
    {
        bourne::json value = object[user_idKey];


        

        AnyType* obj = &user_id;
		obj->fromJson(value.dump());

    }

    const char *userKey = "user";
    
    if(object.has_key(userKey)) 
    {
        bourne::json value = object[userKey];


        

        AnyType* obj = &user;
		obj->fromJson(value.dump());

    }


}

bourne::json
EmojiReaction_allOf::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["emoji_code"] = getEmojiCode().toJson();

    




	object["emoji_name"] = getEmojiName().toJson();

    




	object["reaction_type"] = getReactionType().toJson();

    




	object["user_id"] = getUserId().toJson();

    




	object["user"] = getUser().toJson();


    return object;

}

AnyType
EmojiReaction_allOf::getEmojiCode()
{
	return emoji_code;
}

void
EmojiReaction_allOf::setEmojiCode(AnyType  emoji_code)
{
	this->emoji_code = emoji_code;
}

AnyType
EmojiReaction_allOf::getEmojiName()
{
	return emoji_name;
}

void
EmojiReaction_allOf::setEmojiName(AnyType  emoji_name)
{
	this->emoji_name = emoji_name;
}

AnyType
EmojiReaction_allOf::getReactionType()
{
	return reaction_type;
}

void
EmojiReaction_allOf::setReactionType(AnyType  reaction_type)
{
	this->reaction_type = reaction_type;
}

AnyType
EmojiReaction_allOf::getUserId()
{
	return user_id;
}

void
EmojiReaction_allOf::setUserId(AnyType  user_id)
{
	this->user_id = user_id;
}

AnyType
EmojiReaction_allOf::getUser()
{
	return user;
}

void
EmojiReaction_allOf::setUser(AnyType  user)
{
	this->user = user;
}

File: spring/src/main/java/org/openapitools/model/EmojiReactionAllOf.java
package org.openapitools.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * EmojiReactionAllOf
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2021-08-08T21:23:21.107359Z[Etc/UTC]")
public class EmojiReactionAllOf   {
  @JsonProperty("emoji_code")
  private JsonNullable<Object> emojiCode = JsonNullable.undefined();

  @JsonProperty("emoji_name")
  private JsonNullable<Object> emojiName = JsonNullable.undefined();

  @JsonProperty("reaction_type")
  private JsonNullable<Object> reactionType = JsonNullable.undefined();

  @JsonProperty("user_id")
  private JsonNullable<Object> userId = JsonNullable.undefined();

  @JsonProperty("user")
  private JsonNullable<Object> user = JsonNullable.undefined();

  public EmojiReactionAllOf emojiCode(Object emojiCode) {
    this.emojiCode = JsonNullable.of(emojiCode);
    return this;
  }

  /**
   * Get emojiCode
   * @return emojiCode
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getEmojiCode() {
    return emojiCode;
  }

  public void setEmojiCode(JsonNullable<Object> emojiCode) {
    this.emojiCode = emojiCode;
  }

  public EmojiReactionAllOf emojiName(Object emojiName) {
    this.emojiName = JsonNullable.of(emojiName);
    return this;
  }

  /**
   * Get emojiName
   * @return emojiName
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getEmojiName() {
    return emojiName;
  }

  public void setEmojiName(JsonNullable<Object> emojiName) {
    this.emojiName = emojiName;
  }

  public EmojiReactionAllOf reactionType(Object reactionType) {
    this.reactionType = JsonNullable.of(reactionType);
    return this;
  }

  /**
   * Get reactionType
   * @return reactionType
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getReactionType() {
    return reactionType;
  }

  public void setReactionType(JsonNullable<Object> reactionType) {
    this.reactionType = reactionType;
  }

  public EmojiReactionAllOf userId(Object userId) {
    this.userId = JsonNullable.of(userId);
    return this;
  }

  /**
   * Get userId
   * @return userId
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getUserId() {
    return userId;
  }

  public void setUserId(JsonNullable<Object> userId) {
    this.userId = userId;
  }

  public EmojiReactionAllOf user(Object user) {
    this.user = JsonNullable.of(user);
    return this;
  }

  /**
   * Get user
   * @return user
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getUser() {
    return user;
  }

  public void setUser(JsonNullable<Object> user) {
    this.user = user;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    EmojiReactionAllOf emojiReactionAllOf = (EmojiReactionAllOf) o;
    return Objects.equals(this.emojiCode, emojiReactionAllOf.emojiCode) &&
        Objects.equals(this.emojiName, emojiReactionAllOf.emojiName) &&
        Objects.equals(this.reactionType, emojiReactionAllOf.reactionType) &&
        Objects.equals(this.userId, emojiReactionAllOf.userId) &&
        Objects.equals(this.user, emojiReactionAllOf.user);
  }

  @Override
  public int hashCode() {
    return Objects.hash(emojiCode, emojiName, reactionType, userId, user);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class EmojiReactionAllOf {\n");
    
    sb.append("    emojiCode: ").append(toIndentedString(emojiCode)).append("\n");
    sb.append("    emojiName: ").append(toIndentedString(emojiName)).append("\n");
    sb.append("    reactionType: ").append(toIndentedString(reactionType)).append("\n");
    sb.append("    userId: ").append(toIndentedString(userId)).append("\n");
    sb.append("    user: ").append(toIndentedString(user)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

